### PR TITLE
feat(types/interaction): add "resolved" field to DiscordInteractionCommand

### PR DESCRIPTION
**Pull request by itohatweb**
Closes: #593

### CODE REVIEW

**Comment by ayntee** [comment]
Can you confirm if that's the valid related issue?

**Comment by itohatweb** [comment]
> Can you confirm if that's the valid related issue?

Oh sorry, GitHub auto completed my number...

File: src/types/api/interaction.ts [hunk 1]
```diff
@@ -31,8 +36,24 @@ export interface DiscordInteractionData {
...
   /** the name of the invoked command */
   name: string;
+  /** converted users + roles + channels */
+  resolved?: DiscordInteractionDataResolved;
   /** the params + values from the user */
-  options: DiscordInteractionDataOption[];
+  options?: DiscordInteractionDataOption[];
+}
+
+export interface DiscordInteractionDataResolved {
```
Comment (ayntee) [inline src/types/api/interaction.ts:45]:
```suggestion
export interface DiscordApplicationCommandInteractionDataResolved {
```
To be consistent with documentation

Comment (itohatweb) [reply src/types/api/interaction.ts:45]: So 2 options:
1. Keep it as it is now
2. Name it `DiscordApplicationCommandInteractionDataResolved` but also rename all other interaction interfaces

Comment (Skillz4Killz) [reply src/types/api/interaction.ts:45]: I would say option 3 is best, change this one real quick and then change others in another pr in future in v11

**Review by Skillz4Killz — CHANGES_REQUESTED** [review]

File: src/types/api/interaction.ts [hunk 2]
```diff
@@ -31,8 +36,24 @@ export interface DiscordInteractionData {
   id: string;
   /** the name of the invoked command */
   name: string;
+  /** converted users + roles + channels */
+  resolved?: DiscordInteractionDataResolved;
```
Comment (itohatweb) [inline src/types/api/interaction.ts:40]:
```suggestion
  resolved?: DiscordApplicationCommandInteractionDataResolved;
```

**Review by Skillz4Killz — APPROVED** [review]

**Review by ayntee — APPROVED** [review]

**Pull request merged**